include/deal.II/fe/fe_coupling_values.h

```diff
@@ -660,6 +1265,495 @@ namespace FEValuesViews
+{
+  return second_renumbering_data->n_dofs;
+}
+
+
+
+template <int dim1, int dim2, int spacedim>
+inline unsigned int
+FECouplingValues<dim1, dim2, spacedim>::n_quadrature_points() const
+{
+  return n_quadrature_points_;
+}
+
+
+
+template <int dim1, int dim2, int spacedim>
+std::pair<Point<spacedim>, Point<spacedim>>
+FECouplingValues<dim1, dim2, spacedim>::quadrature_point(
+  const unsigned int quadrature_point) const
+{
+  AssertIndexRange(quadrature_point, n_quadrature_points_);
+  auto first_q = first_fe_values->quadrature_point(
+    first_renumbering_data->quadrature_renumbering.empty() ?
+      quadrature_point :
+      first_renumbering_data->quadrature_renumbering[quadrature_point]);
+
+  auto second_q = second_fe_values->quadrature_point(
+    second_renumbering_data->quadrature_renumbering.empty() ?
+      quadrature_point :
+      second_renumbering_data->quadrature_renumbering[quadrature_point]);
+
+  return {first_q, second_q};
+}
+
+
+
+template <int dim1, int dim2, int spacedim>
+std::array<unsigned int, 2>
+FECouplingValues<dim1, dim2, spacedim>::
+  coupling_quadrature_to_quadrature_indices(
+    const unsigned int quadrature_point) const
+{
+  AssertIndexRange(quadrature_point, n_quadrature_points_);
+  const auto first_id =
+    first_renumbering_data->quadrature_renumbering.empty() ?
+      quadrature_point :
+      first_renumbering_data->quadrature_renumbering[quadrature_point];
+
+  const auto second_id =
+    second_renumbering_data->quadrature_renumbering.empty() ?
+      quadrature_point :
+      second_renumbering_data->quadrature_renumbering[quadrature_point];
+  return {first_id, second_id};
+}
+
+
+
+template <int dim1, int dim2, int spacedim>
+std::array<unsigned int, 2>
+FECouplingValues<dim1, dim2, spacedim>::coupling_dof_to_dof_indices(
+  const unsigned int coupling_dof_index) const
+{
+  AssertIndexRange(coupling_dof_index, n_coupling_dofs_);
+  const auto first_id =
+    first_renumbering_data->dof_renumbering.empty() ?
+      coupling_dof_index :
+      first_renumbering_data->dof_renumbering[coupling_dof_index];
+
+  const auto second_id =
+    second_renumbering_data->dof_renumbering.empty() ?
+      coupling_dof_index :
+      second_renumbering_data->dof_renumbering[coupling_dof_index];
+  return {first_id, second_id};
+}
+
+
+
+template <int dim1, int dim2, int spacedim>
+template <typename Extractor>
+inline const typename FECouplingValues<dim1, dim2, spacedim>::
+  template FirstCoupling<Extractor>
+  FECouplingValues<dim1, dim2, spacedim>::first(
+    const Extractor &extractor) const
+{
+  return FirstCoupling<Extractor>(extractor);
+}
+
+
+
+template <int dim1, int dim2, int spacedim>
+template <typename Extractor>
+inline const typename FECouplingValues<dim1, dim2, spacedim>::
+  template SecondCoupling<Extractor>
+  FECouplingValues<dim1, dim2, spacedim>::second(
+    const Extractor &extractor) const
+{
+  return SecondCoupling<Extractor>(extractor);
+}
```

Comment:
I missed this in the class declaration. There is no point returning `const` object that is not a reference. You might as well remove the `const`.